oml/models/vit/clip.py

```diff
@@ -0,0 +1,240 @@
+from collections import OrderedDict
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import torch
+from torch import nn
+
+from oml.interfaces.models import IExtractor
+from oml.models.utils import remove_prefix_from_state_dict
+from oml.utils.io import download_checkpoint
+
+CLIP_MODELS = {
```

Comment:
can we follow the same structure as we have in ViT?

when we have a set of functions-constructors and the parameters related to the architecture exist only inside these functions

let's discuss it offline

Reply:
It would be also good to have the same `pretrained_models` field in that class